components/tikv_util/src/resizable_threadpool.rs

```diff
@@ -1,161 +1,332 @@
 // Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
 
-use std::sync::{Arc, RwLock};
+use std::{
+    collections::HashSet,
+    hash::{Hash, Hasher},
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc, Mutex,
+    },
+    time::Duration,
+};
 
 use futures::Future;
-use tokio::{io::Result as TokioResult, runtime::Runtime};
-
-/// DaemonRuntime is a "background" runtime, which contains "daemon" tasks:
-/// any task spawn into it would run until finish even the runtime isn't
-/// referenced.
-pub struct DaemonRuntime(Option<Runtime>);
-
-impl DaemonRuntime {
-    /// spawn a daemon task to the runtime.
-    pub fn spawn(self: &Arc<Self>, f: impl Future<Output = ()> + Send + 'static) {
-        let wkr = self.clone();
-        self.0.as_ref().unwrap().spawn(async move {
-            f.await;
-            drop(wkr)
-        });
-    }
+use tokio::{
+    io::Result as TokioResult,
+    runtime::{Builder, Handle, Runtime},
+    time::interval,
+};
+
+struct RcRuntime {
+    inner: Option<Runtime>,
```

Comment:
I refactored the structure to make it safe to use handle() now.
Only the `ResizableRuntime` could create handle, and it will ensure the lifecycle of it.